BLOB_STORE_DESIGN.md

```diff
@@ -0,0 +1,45 @@
+# Blob Store
+
+The blob store is a mechanism to store arbitrary data in Kapacitor.
+The data stored is immutable and opaque to Kapacitor.
+
+Data is stored as blobs.
+A naming system is used to refer various blobs within the store.
+Names can be modified and the history of a name/blob associations are recorded.
```

Comment:
What does it mean for names to be "modified"? Is there a double-indirection happening to look up the current blob for a name?